### ISSUE TITLE
When input is torch.Tensor, results for shape batch_dim are overridden with "-1"

### ISSUE THREAD

**Issue by fxmarty** [TylerYep/torch-summary#12]
Hello,

Thank you for updating the original torchsummary.

See this line for the issue I think: https://github.com/TylerYep/torch-summary/blob/784c5b7995925826bb98f532d2af965ecd9ca246/torchsummary/layer_info.py#L55

Here is my input with 32 being the batch size:

```
data = torch.zeros((32,2,220500)).to(device)
summary(umx, input_data=data,batch_dim=0)
```

Here's an example of output I get with my model:

```
------------------------------------------------------------------------------------------
Layer (type:depth-idx)                   Output Shape              Param #
==========================================================================================
├─Sequential: 1-1                        [-1, 32, 2, 2049]         --
├─STFT: 1-2                              [-1, 2, 2049, 212, 2]     (4,096)
├─Sequential: 1                          []                        --
|    └─STFT: 2-1                         [-1, 2, 2049, 212, 2]     (recursive)
├─Spectrogram: 1-3                       [-1, 32, 2, 2049]         --
├─Sequential: 1                          []                        --
|    └─Spectrogram: 2-2                  [-1, 32, 2, 2049]         --
├─Linear: 1-4                            [-1, 512]                 1,037,312
├─BatchNorm1d: 1-5                       [-1, 512]                 1,024
├─LSTM: 1-6                              [-1, 32, 512]             4,730,880
├─Linear: 1-7                            [-1, 512]                 524,288
├─BatchNorm1d: 1-8                       [-1, 512]                 1,024
├─Linear: 1-9                            [-1, 4098]                2,098,176
├─BatchNorm1d: 1-10                      [-1, 4098]                8,196
==========================================================================================
```

As you can see, for each layer output, the dimension zero is overridden with -1.

If I plug for example
```
data = torch.zeros((32,2,220500)).to(device)
summary(umx, input_data=data,batch_dim=1)
```

Then the output is
```
------------------------------------------------------------------------------------------
Layer (type:depth-idx)                   Output Shape              Param #
==========================================================================================
├─Sequential: 1-1                        [212, -1, 2, 2049]        --
├─STFT: 1-2                              [32, -1, 2049, 212, 2]    (4,096)
├─Sequential: 1                          []                        --
|    └─STFT: 2-1                         [32, -1, 2049, 212, 2]    (recursive)
├─Spectrogram: 1-3                       [212, -1, 2, 2049]        --
├─Sequential: 1                          []                        --
|    └─Spectrogram: 2-2                  [212, -1, 2, 2049]        --
├─Linear: 1-4                            [6784, -1]                1,037,312
├─BatchNorm1d: 1-5                       [6784, -1]                1,024
├─LSTM: 1-6                              [212, -1, 512]            4,730,880
├─Linear: 1-7                            [6784, -1]                524,288
├─BatchNorm1d: 1-8                       [6784, -1]                1,024
├─Linear: 1-9                            [6784, -1]                2,098,176
├─BatchNorm1d: 1-10                      [6784, -1]                8,196
==========================================================================================
```

So it seems to me there is a problem there. The problem may not come from my model since [pytorch-model-summary](https://pypi.org/project/pytorch-model-summary/) yields good results.

Am I mistaken somewhere?

Cheers,

**Comment by TylerYep** [comment]
Hmm, yes I think this is an issue. For context, in previous iterations of torch-summary, we did not allow the user to pass in a tensor input_data. Since we only allowed input sizes, we needed to infer the batch dimension, and create a tensor with batch size 2 in order to run forward() on the model.

In torch-summary 1.5.0, we will need to modify the API such that if input_data is a tensor, we will use and show the given batch dimension. If input_data is a size, we will use the batch dimension if it is given, or we will need to add some flag specifying the batch has not been provided, and we will infer it.

If input_data is not provided, we do not attempt a forward pass.

Thank you for bringing this issue up - it should be resolved within the next week.
